storage-dynamodb/src/main/java/io/delta/storage/DynamoDBLogStore.java

```diff
@@ -16,10 +16,242 @@
+import com.amazonaws.services.dynamodbv2.model.ExpectedAttributeValue;
+import com.amazonaws.services.dynamodbv2.model.GetItemRequest;
+import com.amazonaws.services.dynamodbv2.model.KeySchemaElement;
+import com.amazonaws.services.dynamodbv2.model.KeyType;
+import com.amazonaws.services.dynamodbv2.model.ProvisionedThroughput;
+import com.amazonaws.services.dynamodbv2.model.PutItemRequest;
+import com.amazonaws.services.dynamodbv2.model.QueryRequest;
+import com.amazonaws.services.dynamodbv2.model.ResourceNotFoundException;
+import com.amazonaws.services.dynamodbv2.model.ResourceInUseException;
+import com.amazonaws.services.dynamodbv2.model.ScalarAttributeType;
+import com.amazonaws.auth.AWSCredentialsProvider;
+
+import com.amazonaws.regions.Region;
+import com.amazonaws.regions.Regions;
+
+import java.util.NoSuchElementException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
 public class DynamoDBLogStore extends BaseExternalLogStore {
-    public DynamoDBLogStore(Configuration hadoopConf) {
+    private static final Logger LOG = LoggerFactory.getLogger(DynamoDBLogStore.class);
+
+    private AmazonDynamoDBClient client = null;
+    private String tableName;
+    private String credentialsProviderName;
+    private String regionName;
+    private String endpoint;
+
+    public DynamoDBLogStore(Configuration hadoopConf) throws IOException {
         super(hadoopConf);
+        tableName = getParam(hadoopConf, "tableName", "delta_log");
+        credentialsProviderName = getParam(
+            hadoopConf,
+            "credentials.provider",
+            "com.amazonaws.auth.DefaultAWSCredentialsProviderChain"
+        );
+        regionName = getParam(hadoopConf, "region", "us-east-1");
+        endpoint = getParam(hadoopConf, "endpoint", null);
+        client = getClient();
+        tryEnsureTableExists(hadoopConf);
+    }
+
+   /*
+    * Write to db in exclusive way.
+    * Method should throw @java.nio.file.FileAlreadyExistsException if path exists in cache.
+    */
+    @Override
+    protected void putExternalEntry(
+        ExternalCommitEntry entry, boolean overwrite
+    ) throws IOException {
+        try {
+            LOG.debug(String.format("putItem %s, overwrite: %s", entry, overwrite));
+            client.putItem(createPutItemRequest(entry, tableName, overwrite));
+        } catch (ConditionalCheckFailedException e) {
+            LOG.debug(e.toString());
+            throw new java.nio.file.FileAlreadyExistsException(
+                entry.absoluteJsonPath().toString()
+            );
+        }
+    }
+
+    @Override
+    protected ExternalCommitEntry getExternalEntry(
+        Path absoluteTablePath, Path absoluteJsonPath
+    ) throws IOException {
+        Map<String, AttributeValue> attributes = new ConcurrentHashMap<>();
+        attributes.put("tablePath", new AttributeValue(absoluteTablePath.toString()));
+        attributes.put("fileName", new AttributeValue(absoluteJsonPath.toString()));
```

Comment:
@mrk-its in this `get` call. fileName is the absoluteJsonPath?
yet in the `put` method, and in `createPutItemRequest` you use `entry.fileName` ?

Reply:
this seems wrong to me

Reply:
It seems parameter name is wrong and we are passing to relative path here, but I need to confirm it

Reply:
```
Path prevPath = deltaFile(tablePath, prevVersion);
ExternalCommitEntry entry = getExternalEntry(tablePath, prevPath);
```

Seems like you pass in the full path

Reply:
```
PutItemRequest createPutItemRequest(
...
    attributes.put("fileName", new AttributeValue(entry.fileName));
```

Ya this is concerning to me

Reply:
@scottsand-db good catch! This is definitely a bug. I was thinking a bit how it worked in integration test with error injection. It seems fixing previous log entries during write was broken because of that - (as we always had null here: https://github.com/delta-io/delta/pull/1023/files#diff-f988aaca0fb8150f14791ddae6cdb5cc62670b78114dd3d4942ef1192f9926d2R100) But everything worked because these incomplete entries were fixed in `listFrom` call probably. I'll create separate PR with bugfix.